_posts/2020-06-05-mirror-maker-2-eventhub.md

```diff
@@ -0,0 +1,290 @@
+    bootstrapServers: <eventhubs-namespace>.servicebus.windows.net:9093
+    config:
+      config.storage.replication.factor: 1
+      offset.storage.replication.factor: 1
+      status.storage.replication.factor: 1
+      producer.connections.max.idle.ms: 180000
+      producer.metadata.max.age.ms: 180000
+    authentication:
+      type: plain
+      username: $ConnectionString
+      passwordSecret:
+        secretName: eventhubssecret
+        password: eventhubspassword
+    tls:
+      trustedCertificates: []
+  mirrors:
+  - sourceCluster: "eventhub"
+    targetCluster: "my-cluster"
+    sourceConnector:
+      config:
+        replication.factor: 1
+        offset-syncs.topic.replication.factor: 1
+        sync.topic.acls.enabled: "false"
+    heartbeatConnector:
+      config:
+        heartbeats.topic.replication.factor: 1
+    checkpointConnector:
+      config:
+        checkpoints.topic.replication.factor: 1
+    topicsPattern: ".*"
+    groupsPattern: ".*"
+```
+
+Create the `KafkaMirrorMaker2` on the Kubernetes cluster saving the above resource into a `kafka-mirror-maker-2-from-eh.yaml` file.
+The Strimzi cluster operator takes care of it deploying Kafka Mirror Maker 2 using the above configuration.
+
+```shell
+kubectl apply -f kafka-mirror-maker-2-from-eh.yaml -n kafka
+```
+
+In this way, we can add another producer on the Event Hub side sending messages to the `testeh` event hub.
+We can use the `kafka-console-producer` for this purpose.
+
+```shell
+bin/kafka-console-producer.sh --bootstrap-server functiontesteh.servicebus.windows.net:9093 --topic testeh --producer.config kafka_eventhub.properties
+If you don't see a command prompt, try pressing enter.
+> "Hello from Strimzi Mirror Maker 2 sent to testeh Event Hub
+>
+```
+
+The producer is in addition to the one we used in the previous paragraph which is producing locally to `testeh` in the Kafka cluster.
+
+```shell
+kubectl -n kafka run kafka-producer -ti --image=strimzi/kafka:0.18.0-kafka-2.5.0 --rm=true --restart=Never -- bin/kafka-console-producer.sh --broker-list my-cluster-kafka-bootstrap:9092 --topic testeh
+If you don't see a command prompt, try pressing enter.
+>"Hello from Strimzi Mirror Maker 2 sent to testeh Kafka"
+>
+```
+
+We can also add a consumer on the local Kafka cluster getting messages on the corresponding mirrored `eventhub.testeh` topic.
+
+```shell
+kubectl -n kafka run kafka-consumer -ti --image=strimzi/kafka:0.18.0-kafka-2.5.0 --rm=true --restart=Never -- bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --whitelist .*testeh
+"Hello from Strimzi Mirror Maker 2 sent to testeh Event Hub"
+"Hello from Strimzi Mirror Maker 2 sent to testeh Kafka"
+```
+
+The consumer used in the previous paragrah will get both the messages as well.
+
+```shell
+bin/kafka-console-consumer.sh --bootstrap-server <eventhubs-namespace>.servicebus.windows.net:9093 --topic testeh --consumer.config kafka_eventhub.properties
```

Comment:
This one does not seem to use pattern as suggested in the next paragraph.

Reply:
You know what, right? :P